clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java

```diff
@@ -318,11 +320,67 @@ public FileStatus getFileStatus(Path path) throws IOException {
 
     FileStatus ret = new FileStatus(fileStatus.getLength(), fileStatus.isIsFolder(),
         BLOCK_REPLICATION_CONSTANT, fileStatus.getBlockSizeBytes(), fileStatus.getCreationTimeMs(),
-        fileStatus.getCreationTimeMs(), null, null, null, new Path(mTachyonHeader + tPath));
+            fileStatus.getCreationTimeMs(), new FsPermission((short) fileStatus.getPermission()),
+            fileStatus.getUserName(), fileStatus.getGroupName(), new Path(mTachyonHeader + tPath));
     return ret;
   }
 
   /**
+   * Changes owner or group of a path (i.e. a file or a directory). If username is null, the
+   * original username remains unchanged. Same as groupname. If username and groupname are non-null,
+   * both of them will be changed.
+   *
+   * @param path path to set owner or group
+   * @param username username to be set
+   * @param groupname groupname to be set
+   * @throws IOException if changing owner or group of the path failed
+   */
+  @Override
+  public void setOwner(Path path, final String username, final String groupname)
+      throws IOException {
+    TachyonURI tPath = new TachyonURI(Utils.getPathWithoutScheme(path));
+    Path hdfsPath = Utils.getHDFSPath(tPath, mUnderFSAddress);
+    LOG.info("setOwner({},{},{}) HDFS Path: {} Tachyon Path: {}{}", path, username, groupname,
+        hdfsPath, mTachyonHeader, tPath);
+    try {
+      if (username != null && !username.isEmpty()) {
```

Comment:
nit: you could create a single options object and use a single RPC for both username and group permissions.